contract/contracts/Voting/Voting.sol

```diff
@@ -0,0 +1,107 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+import '@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol';
+import '@openzeppelin/contracts-upgradeable/utils/PausableUpgradeable.sol';
+import '@openzeppelin/contracts-upgradeable/utils/ReentrancyGuardUpgradeable.sol';
+import "./Types.sol";
+
+contract Voting is OwnableUpgradeable, PausableUpgradeable, ReentrancyGuardUpgradeable {
+
+    mapping(uint256 => Types.Proposal) public proposals;
+    uint256 public proposalCount;
+
+    event ProposalAdded(uint256 indexed proposalId, string title, string description);
+    event DirectiveVoted(address indexed voter, uint256 indexed proposalId, uint256 vote);
+    event ElectionVoted(address indexed voter, uint256 indexed proposalId, address indexed candidateAddress);
+    event ProposalConcluded(uint256 indexed proposalId, Types.Proposal proposal);
+
+    function initialize() public initializer {
+        __Ownable_init(msg.sender);
+    }
+
+   function addProposal(Types.Proposal calldata _proposal) public onlyOwner whenNotPaused {
+        proposals[proposalCount] = _proposal;
+        emit ProposalAdded(proposalCount, _proposal.title,_proposal.description);
+        proposalCount++;
+    }
+
+    function getProposals() public view returns (Types.Proposal[] memory) {
+        Types.Proposal[] memory _proposals = new Types.Proposal[](proposalCount);
+        for (uint256 i = 0; i < proposalCount; i++) {
+            _proposals[i] = proposals[i];
+        }
+        return _proposals;
+    }
```

Comment:
i think we dont need to do this, instead we can read from event `ProposalAdded`

Reply:
and instead of this function we create get one proposal by `proposalId`

Reply:
That event is emitted only when a proposal is added, how do we retrieve all proposals to display?

Reply:
I'm thinking we'll return all proposals and then apply a filter